server/routes/index.js

```diff
@@ -5,12 +5,15 @@ const getCardHandler = require('./getCard');
 const deleteCardHandler = require('./deleteCard');
 const clearCardHandler = require('./clearCard');
 
+const { getProduct } = require('../controllers/index');
+
 const router = express.Router();
 
 router.post('/singup', addUserHandler);
 router.post('/addcard', addCardHandler);
-router.post('/getcard', getCardHandler);
-router.post('/clearcard', getCardHandler);
-router.post('/deletecard', getCardHandler);
+router.get('/getcard/:userId', getCardHandler);
+router.delete('/clearcard/:userId', clearCardHandler);
+router.delete('/deletecard/:userId/:productId', deleteCardHandler);
```

Comment:
You shouldn't send the userId for these routes by params, you should take it from the cookies after verifying, even if you still didn't handle this middleware properly you should add it and make your controller prepared for it
```suggestion
router.get('/getcard/:userId', getCardHandler);
router.delete('/clearcard', clearCardHandler);
router.delete('/deletecard/:productId', deleteCardHandler);
```